src/javascripts/jquery.tipsy.js

```diff
@@ -74,35 +74,52 @@
                     height: this.$element[0].getBoundingClientRect().height
                 });
                 
-                var actualWidth = $tip[0].offsetWidth,
-                    actualHeight = $tip[0].offsetHeight,
-                    gravity = maybeCall(this.options.gravity, this.$element[0]);
-                
-                var tp;
+                var tipCss = {};
+                var actualWidth = $tip[0].offsetWidth;
+                var gravity = maybeCall(this.options.gravity, this.$element[0]);
+
+                if (gravity.length === 2) {
+                    if (gravity.charAt(1) === 'w') {
+                        tipCss.left = pos.left + pos.width / 2 - 15;
+                    } else {
+                        tipCss.left = pos.left + pos.width / 2 - actualWidth + 15;
+                    }
+                }
+
+                var actualHeight = $tip[0].offsetHeight;
+
                 switch (gravity.charAt(0)) {
                     case 'n':
-                        tp = {top: pos.top + pos.height + this.options.offset, left: pos.left + pos.width / 2 - actualWidth / 2};
+                        // left could already be set if gravity is 'nw' or 'ne'
+                        if (typeof tipCss.left === 'undefined') {
+                            tipCss.left = pos.left + pos.width / 2 - actualWidth / 2;
+                        }
+                        tipCss.top = pos.top + pos.height + this.options.offset;
                         break;
                     case 's':
-                        tp = {top: pos.top - actualHeight - this.options.offset, left: pos.left + pos.width / 2 - actualWidth / 2};
+                        // left could already be set if gravity is 'sw' or 'se'
+                        if (typeof tipCss.left === 'undefined') {
+                            tipCss.left = pos.left + pos.width / 2 - actualWidth / 2;
```

Comment:
It looks like this sets `tipCss.left` assuming that the gravity is `se`, is this correct? (From above, it looks like `*w` gravity would have a different value for `tipCss.left`)